File: web/src/main/java/no/nav/modiapersonoversikt/service/ansattservice/domain/AnsattEnhet.java
package no.nav.modiapersonoversikt.service.ansattservice.domain;

import java.io.Serializable;
import java.util.function.Function;

import static org.apache.commons.lang3.StringUtils.upperCase;

public class AnsattEnhet implements Serializable {

    public final String enhetId;
    public final String enhetNavn;
    public final String status;

    public AnsattEnhet(String enhetId, String enhetNavn) {
        this(enhetId, enhetNavn, null);
    }

    public AnsattEnhet(String enhetId, String enhetNavn, String status) {
        this.enhetId = enhetId;
        this.enhetNavn = enhetNavn;
        this.status = status;
    }

    public boolean erAktiv() {
        return "AKTIV".equals(upperCase(status));
    }

    public static final Function<AnsattEnhet, String> TIL_ENHET_ID = ansattEnhet -> ansattEnhet.enhetId;
}

File: web/src/main/java/no/nav/modiapersonoversikt/service/ansattservice/domain/AnsattEnhet.kt
package no.nav.modiapersonoversikt.service.ansattservice.domain

import java.util.function.Function

data class AnsattEnhet(
    val enhetId: String,
    val enhetNavn: String,
    val status: String? = null,
) {
    fun erAktiv(): Boolean = status?.uppercase() == "AKTIV"

    companion object {
        val TIL_ENHET_ID: Function<AnsattEnhet?, String?> =
            Function { ansattEnhet: AnsattEnhet? -> ansattEnhet!!.enhetId }
    }
}
